components/brave_rewards/browser/rewards_service_observer.h

```diff
@@ -61,6 +64,13 @@ class RewardsServiceObserver : public base::CheckedObserver {
   virtual void OnPublisherListNormalized(
       RewardsService* rewards_service,
       brave_rewards::ContentSiteList list) {};
+  virtual void OnAdsNotificationsData(brave_rewards::RewardsService* rewards_service,
+                              int total_viewed,
+                              double estimated_earnings) {};
+  virtual void OnAdsNotificationsData(
+      brave_rewards::RewardsService* rewards_service,
+      int total_viewed,
+      double estimated_earnings) {}
   // DO NOT ADD ANY MORE METHODS HERE UNLESS IT IS A BROADCAST NOTIFICATION
```

Comment:
comment re above

Reply:
Strange in my local copy I do not see this dupe?

Reply:
@bridiver Ryan spoke with Nejc and it was decided to revert changes which removed the client side polling (calling GetAdNotificationHistory). Speaking with Ryan the observer is needed to listen for notifications once they have been received but we also need the client side polling to detect whether we need to update data on the rewards settings page. @bridiver @NejcZdovc can you please let Ryan know how this should be resolved.

Reply:
if we need to know when changes happen that is when the observer should trigger. We shouldn't poll client-side, we should push from ledger and then fan out to observers in RewardsServiceImpl

Reply:
@bridiver let's say I have `brave://rewards` closed when a notification was shown (observer triggered, but doesn't hit the rewards settings reducer) then I open the page. If I can't poll, how do I get the page to refresh its data?

Reply:
so we need both. We need client poll when user visits settings page (could be closed for a week and we need to get the latest data) and we need to update settings page when user have this page opened and is viewing ad. So we need callback logic and observer logic. Will modify code to do both this things.